src/runtime/particle-execution-context.ts

```diff
@@ -195,19 +195,24 @@ export class ParticleExecutionContext {
     };
   }
 
-  async _instantiateParticle(id: string, spec: ParticleSpec, proxies: Map<string, StorageProxy>) {
+  // tslint:disable-next-line: no-any
+  async _instantiateParticle(id: string, spec: ParticleSpec, proxies: Map<string, StorageProxy>): Promise<[any, () => Promise<void>]> {
     let resolve : () => void = null;
     const p = new Promise<void>(res => resolve = res);
     this.pendingLoads.push(p);
     const clazz = await this.loader.loadParticleClass(spec);
     const capabilities = this.defaultCapabilitySet();
     const particle = new clazz(); // TODO: how can i add an argument to DomParticle ctor?
-    particle.id = id;
-    particle.capabilities = capabilities;
+
```

Comment:
Please advise on what should be done here..

I tried to move the capabilities into the constructor but then saw permission errors in the console.

Reply:
What are the permission errors you saw? I would have thought moving capabilities into the constructor was the right thing to do too.

Reply:
[ooops, see below for detailed description]